epix/io.py

```diff
@@ -145,27 +151,53 @@ def loader(directory, file_name, arg_debug=False, outer_cylinder=None, kwargs_up
     m = ak.num(interactions['ed']) > 0
     interactions = interactions[m]
 
-    return interactions
+    return interactions, n_simulated_events
+
+
+def _get_ttree(directory, file_name):
+    """
+    Function which searches for the correct ttree in MC root file.
+
+    :param directory: Directory where file is
+    :param file_name: Name of the file
+    :return: root ttree and number of simulated events
+    """
+    root_dir = uproot.open(os.path.join(directory, file_name))
+
+    # Searching for TTree according to old/new MC file structure:
+    if root_dir.classname_of('events') == 'TTree':
+        ttree = root_dir['events']
+        n_simulated_events = root_dir['nEVENTS'].members['fVal']
```

Comment:
Just a reminder that, for this option (MC newer than 4.0.0), we can here fetch the LXe level and pass it to the geometry config, to define properly the gas phase. I will get to it.